cdc/sinkv2/eventsink/cloudstorage/dml_worker.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package cloudstorage
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/br/pkg/storage"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/cdc/sink/codec/common"
+	"github.com/pingcap/tiflow/pkg/chann"
+	"go.uber.org/zap"
+)
+
+// dmlWorker denotes a worker responsible for writing messages to cloud storage.
+type dmlWorker struct {
+	// worker id
+	id            int
+	changeFeedID  model.ChangeFeedID
+	storage       storage.ExternalStorage
+	flushInterval time.Duration
+	// msgChannels maintains a mapping of <table, unbounded message channel>.
+	msgChannels sync.Map
```

Comment:
One unbounded message channel is accompanied with a goroutine. If there are so many tables, this may be an issue.